sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1495,6 +1495,53 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
   return PI_SUCCESS;
 }
 
+// sub-sub-device
+// TODO: do we need to gather Ordinals or just the SubSubDevicesCount will do ?
+pi_result getCmdQueueOrdinals(pi_device PiSubDevice, pi_uint32 &SubSubDevicesCount, std::vector<int>& Ordinals,
+                              std::vector<ze_command_queue_group_properties_t>& AllQueueProperties) {
+  uint32_t numQueueGroups = 0;
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, nullptr));
+  if (numQueueGroups == 0) {
+    return PI_ERROR_UNKNOWN;
+  }
+  std::vector<ze_command_queue_group_properties_t> QueueProperties(
+      numQueueGroups);
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, QueueProperties.data()));
+
+  SubSubDevicesCount = numQueueGroups;
+  AllQueueProperties = QueueProperties;
+
+  bool noComputeEngineFlag = true;
+  for (uint32_t i = 0; i < numQueueGroups; i++) {
+    if (QueueProperties[i].flags &
+        ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) {
+      Ordinals.push_back(i);
+      noComputeEngineFlag = false;
+    }
+  }
+
+  if (noComputeEngineFlag) {
+    return PI_ERROR_UNKNOWN;
+  }
+
+  return PI_SUCCESS;
+}
+
+pi_result initializeWithOrdinal(pi_device PiSubSubDevice, int Ordinal, std::vector<ze_command_queue_group_properties_t>& QueueProperties) {
```

Comment:
why don't generalize pi_device->initialize() to take optional ordinal?

Reply:
will do

Reply:
done.